.github/instructions/typescript.instructions.md

```
@@ -1,12 +1,17 @@
-- Use TypeScript for all new code
+- Do not prefix interfaces with `I` (e.g., use `User` instead of `IUser`).
 - Follow functional programming principles where possible
-- Use interfaces for data structures and type definitions
 - Prefer immutable data (`const`, `readonly`)
-- Use optional chaining (`?.`) and nullish coalescing (`??`) operators
+- Support `exactOptionalPropertyTypes` by explicitly including `| undefined` in
+  optional property definitions, e.g. `readonly foo?: string | undefined`.
+- Use interfaces for data structures and type definitions
+- When a function only takes one parameter and that parameter is an untyped
+  object, name that parameter's interface `Options`.
+  ```ts
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)